client-spec.md

```diff
@@ -129,27 +135,60 @@ They can do so in any manner. Signers SHOULD collate this data in the Sigstore w
-
-*Transparency.* The Signer SHOULD upload signing metadata to the Transparency Service, but MAY choose to skip this step (for instance, for privacy reasons). In this case, the Signer MUST use a Timestamping Service to provide a timestamp for the signature.
-
-*Other workflows.* A client may support signing workflows different from that described above. For instance, a Signer may want to use a long-lived signing key without an associated certificate; in this case, they can skip the authentication, key generation, and certificate issuance steps. A Signer may have a distinct Certificate Authority. Details for these workflows are out-of-scope for this document.
+*Authentication System.* The signer MUST use an Authentication System supported
+by the Fulcio instance with which they can authenticate.
+
+*Digital signature algorithm.* The signer must choose a digital signature
+algorithm for key generation and signing from the registry (see [Spec: Sigstore
+Registries](./algorithm-registry.md)). The algorithm MUST be in the
+`supportedSigningAlgorithms` of both the Fulcio and Transparency Service
+instances.
+
+*Signature metadata format*. The signature metadata format MUST be in the list
+of `supportedMetadataFormats` in the Transparency Service configuration. This
+list can include both common registry formats (see [Spec: Sigstore
+Registries](./algorithm-registry.md)) or additional plug-in formats. Details
+about plug-in formats are conveyed out-of-band.
+
+The metadata format chosen SHOULD depend on the artifact to sign (some formats
+encode extra metadata about specific artifact types), size (some formats require
+the full artifact; others allow the payload to be hashed), or compatibility with
+other systems.
+
+*Payload pre-hashing.* Some metadata formats store a hash of the payload. In
+this case, the signature is over the *hashed* payload, so that the Transparency
+Service can validate the signature.
+
+In such cases, the Signer must choose a hash algorithm from the registry (see
+[Spec: Sigstore Registries](./algorithm-registry.md)); this algorithm MUST be in
+the `supportedHashAlgorithms` for the Transparency Service.
+
+*Long-lived signing keys.* The Signer may have a pre-existing, long-lived
+signing key with which they would like to sign payloads. This key MUST use an
+algorithm in the `supportedSigningAlgorithms` of both the Fulcio configuration
+and Transparency Service configuration.
+
+In such cases, the Signer can skip the key generation step; the signing
+procedure is otherwise unaltered.
+
+*Timestamping.* Currently, the Transparency Service includes a timestamp in its
+response to the Signer. This timestamp comes from the Transparency Service’s
+internal clock, which is not externally verifiable or immutable. For this
+reason, a Signer SHOULD get their signatures timestamped. However, a Signer MAY
+choose to omit the timestamping step; in this case, the Signer MUST use the
```

Comment:
While this is true now, we should plan for how this should be worded once Rekor V2 is out, which will not provide any signed timestamp in the response.

Reply:
Yeah, I'll make an issue for it!

Reply:
Created this https://github.com/sigstore/architecture-docs/issues/48